src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatibilityAnalyzerTests.cs

```diff
@@ -3198,6 +3198,232 @@ internal static class UnsupportedWindows6_10_Ios_1_5_11
+    {   // This call site is reachable on: 'windows' from version 7.0 to 9.0. 'Test.SupportedOnWindows10()' is only supported on: 'windows' 10.0 and later.
+        [|SupportedOnWindows10()|]; // Warning; calling from Windows 7-8 is still supported but it will fail
+    }
+    [UnsupportedOSPlatform(""windows6.0"")]
+    void NotSupportedAnywhere()
+    {
+        SupportedOnWindows10(); // No warning; no valid call sites
+    }
+    [SupportedOSPlatform(""windows10.0"")]
+    void SupportedOnWindows10() { }
+}";
+            await VerifyAnalyzerAsyncCs(source, s_msBuildPlatforms);
+        }
+
+        [Fact]
+        public async Task ChildCanNarrowUpperLevelSupportedPlatforms()
+        {
+            var source = @"
+using System.Runtime.Versioning;
+
+[assembly:SupportedOSPlatform(""windows"")]
+[assembly:SupportedOSPlatform(""ios"")]
+public class Test
+{
+    [SupportedOSPlatform(""windows"")]
+    private void WindowsOnly() { }
+
+    [SupportedOSPlatform(""ios"")]
+    private void IosOnly () { }
+
+    private void NoAttribute () { }
+
+    public void M1 ()
+    {
+        [|WindowsOnly()|];
+        [|IosOnly()|];
+        NoAttribute();
+    }
+    [SupportedOSPlatform(""Windows"")]
+    public void M2 ()
+    {
+       WindowsOnly();
+       [|IosOnly()|];
+        NoAttribute();
+    }
+    [SupportedOSPlatform(""iOS"")]
+    public void M3 ()
+    {
+        [|WindowsOnly()|];
+        IosOnly();
+        NoAttribute();
+    }
+}";
+            await VerifyAnalyzerAsyncCs(source, s_msBuildPlatforms);
+        }
+
+        [Fact]
+        public async Task EmptyCallsiteReferencesSupportedUnsupportedApisNotWarns()
+        {
+            var source = @"
+using System;
+using System.Runtime.Versioning;
+
+[SupportedOSPlatform(""browser"")]
+[SupportedOSPlatform(""windows"")]
+[SupportedOSPlatform(""macos"")]
+class Test
+{
+    [UnsupportedOSPlatform(""browser"")]
+    private static void ApiUnsupportedOnBrowser () { }
+
+    [SupportedOSPlatform(""macos"")]
+    class TestMacOs // Inside only mac accessable
+    {
+        [SupportedOSPlatform(""windows"")]
+        void WindowsOnly() // The method attribute causes not valid call sites; no warnings no matter what
+        {
+            Test.ApiUnsupportedOnBrowser();
+            Console.Beep(10, 20); // Windows only API
+        }
+    }
+
+    void MethodHasNoAttribute()
+    {
+        [|ApiUnsupportedOnBrowser()|]; // This call site is reachable on: 'macos', 'browser', 'windows'. 'Test.ApiUnsupportedOnBrowser()' is unsupported on: 'browser'.
+        [|Console.Beep(10, 20)|];      // This call site is reachable on: 'macos', 'browser', 'windows'. 'Console.Beep(int, int)' is only supported on: 'windows'.
+    }
+
+    [SupportedOSPlatform(""WINDOWS"")]
+    void WindowsOnly() // Nothing should warn
+    {
+        ApiUnsupportedOnBrowser();
+        Console.Beep(10, 20);
+    }
+
+    [SupportedOSPlatform(""Linux"")] // Not valid because Linux is not in parent list of platforms
```

Comment:
Why is this the case? If I had a type such that only one of the method supported a platform, why would I be required to add the supported attribute on the whole type as opposed to just the method?

Reply:
Right, as you mentioned it is not a meaningful or realistic scenario, but it is happening in multitargeted builds for runtime repo. We have assemblies for multiple platforms where some APIs annotated as platform-specific, the assembly doesn't have any platform attributes but it does build for each platform specified: 
```xml
 <TargetFrameworks>net5.0-windows;net5.0-Unix;net5.0-tvOS;net5.0-Browser</TargetFrameworks>
```
For example, when it builds for windows `net5.0-windows` target MSBuild adds assembly-level [SupportedOSPlatform("windows")] attributes then the individual APIs having different platforms support would produce this scenario and we see warnings for those targeted builds, this is the main reason we are doing this change